### PR TITLE
Add C# 9 test cases (9 rules)

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Related to #3668

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp9.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
+﻿using System.Diagnostics;
+
+string compliant = "compliant";
+string noncompliant = "foobar";
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp9.cs:4]: can you add duplicated literals in the top-level statements?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StaticSealedClassProtectedMembers.CSharp9.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+
+record Record
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StaticSealedClassProtectedMembers.CSharp9.cs:3]: this should be `final` for the rule to work, right?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StaticFieldInGenericClass.CSharp9.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿record StaticFieldInGenericClass<T, U>
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StaticFieldInGenericClass.CSharp9.cs:1]: you can remove the second generic parameter `U`, right?

**Review by andrei-epure-sonarsource — APPROVED** [review]
please address my comments before merging

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StaticSealedClassProtectedMembers.CSharp9.cs (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/StaticSealedClassProtectedMembers.CSharp9.cs:3]: Yes, it should be `sealed`

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**